documentation/add/07-sqw_redesign.md

```diff
@@ -1,58 +1,84 @@
+The class diagram describing the relationship between main Horace classes is provided on the Fig.1:
 
-![Class Overview](../diagrams/sqw-dnd.png)
+![Fig.1. Class Overview](../diagrams/sqw-dnd.png)
 
 
 
 ### SQWDnDBase
 
-Abstract base class for the SQW and DND objects.
+Abstract base class for the `sqw` and `dnd` objects, describing the interface common for both `sqw` and `dnd` objects.
 
-This includes the image data and main header information, available in both the `SQW` and `DnD` classes.
+May include code of the methods with work identically on both `sqw` and `dnd` objects. .
 
-This class will also include common methods, in particular the large number of unary and binary operations which are implemented as calls to operations managers which will be defined in the implementing classes.
+Currently it also include common methods, in particular the large number of unary and binary operations which are implemented as calls to operations managers which will be defined in the implementing classes.
 
 ### SQW
 
-The `SQW` object is the providing the public API to data. Data manipulations are performed on the `PixelData` and `Image` is recalculated.
+The `sqw` object provides the public API to the all relevant experimental data. Main data manipulations are performed on the `PixelData` and `Image` (`dnd` object, within the `sqw` object) is recalculated accordingly.
 
-This class includes the full experiment data including the raw pixel data and details of the instrument and detectors.
+This class includes the full experiment data including the raw pixel data and details of the instrument and detectors. As the `PixelData` containing all information about neutron events is normally very large dataset `sqw` object for some purpoese may be used without `PixelData`. Alternatively, when `PixelData` is large and can not be loaded in memory, the operations on the `PixelData` can be performed on separate data chunks loaded and processed in memory, leaving the main `PixelData` arrays filebased.
+The structure of generic `sqw` object is presented on Fig.2:
 
-![SQW Class Overview](../diagrams/sqw.png)
+![Fig.2. SQW Class Overview](../diagrams/sqw.png)
 
 ### DnDBase, DnD
 
-The `DnD` object is a "cut-down SQW" object containing only `Image` data. This exists in n-dimensional forms, with each class extending an abstract base class. The `PixelData`, `IX_Instr`and `IX_DetectorArray` information is NOT included, and any data manipulation operations are performed directly on the `Image` data.
+The `DnD` object exist as part of  `sqw` object but also can exist as "cut-down `sqw`" object containing only multidimensional `Image` data expressed in the coordinate system of interest. The "Image" exists in 0 to 4 dimensional forms, with each particular number of dimensions class defining the abstract base class. The `PixelData`, `main_header` and `Experiment` information is NOT included, and any data manipulation operations are performed directly on the `Image` data.
```

Comment:
exist->exists
after 'exist as' add 'a separate' ('a' for grammar, 'separate' for style and information)
'defining the abstract base class' is unclear. The abstract base class DnDBase holds the dimension index - concrete subclasses D0D to D4D then fix this.